### PR TITLE
Address/file tracking issues for few more dataflow prototype comments

### PR DESCRIPTION
**Pull request by mavasani**

### CODE REVIEW

File: src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs [hunk 1]
```diff
@@ -1186,11 +1186,6 @@ private IBinaryOperation CreateBoundBinaryOperatorOperation(BoundBinaryOperator
                 (operatorKind == BinaryOperatorKind.ConditionalAnd || operatorKind == BinaryOperatorKind.ConditionalOr) &&
                 operatorMethod?.Parameters.Length == 1)
             {
-                // PROTOTYPE(dataflow): There is no IOperation test coverage for this code besides some Control Flow Graph tests.
```
Comment (mavasani) [inline src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs]: Tracked with https://github.com/dotnet/roslyn/issues/27580

File: src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs [hunk 2]
```diff
@@ -1186,11 +1186,6 @@ private IBinaryOperation CreateBoundBinaryOperatorOperation(BoundBinaryOperator
...
-                // PROTOTYPE(dataflow): Note that sometimes a user-defined conversion can be used to convert left operand to Boolean instead.
```
Comment (mavasani) [inline src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs]: Tracked with https://github.com/dotnet/roslyn/issues/27587

File: src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs [hunk 3]
```diff
@@ -1208,7 +1203,6 @@ private IBinaryOperation CreateBoundBinaryOperatorOperation(BoundBinaryOperator
 
         private IBinaryOperation CreateBoundUserDefinedConditionalLogicalOperator(BoundUserDefinedConditionalLogicalOperator boundBinaryOperator)
         {
-            // PROTOTYPE(dataflow): There is almost no IOperation test coverage for this code besides some Control Flow Graph tests.
```
Comment (mavasani) [inline src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs]: Tracked with https://github.com/dotnet/roslyn/issues/27588

File: src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IThrowOperation.cs [hunk 4]
```diff
@@ -1986,7 +1986,5 @@ void F(System.Exception a, System.Exception b)
 ";
             VerifyFlowGraphForTest<BlockSyntax>(compilation, expectedGraph);
         }
-
-        // PROTOTYPE(dataflow): Port these tests to VB
```
Comment (mavasani) [inline src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IThrowOperation.cs]: Tracked with https://github.com/dotnet/roslyn/issues/27491

File: src/Compilers/Core/Portable/Operations/IFlowCaptureOperation.cs [hunk 5]
```diff
@@ -8,8 +8,6 @@ namespace Microsoft.CodeAnalysis.FlowAnalysis
     /// </summary>
     public interface IFlowCaptureOperation : IOperation
     {
-        // PROTOTYPE(dataflow): Finalize the design how capturing/referencing intermediate results is represented.
```
Comment (mavasani) [inline src/Compilers/Core/Portable/Operations/IFlowCaptureOperation.cs]: We decided to use `CaptureId`. @AlekseyTs let me know if this comment was capturing more work.

File: src/Compilers/Core/Portable/Operations/IFlowCaptureReferenceOperation.cs [hunk 6]
```diff
@@ -9,8 +9,6 @@ namespace Microsoft.CodeAnalysis.FlowAnalysis
     /// </summary>
     public interface IFlowCaptureReferenceOperation : IOperation
     {
-        // PROTOTYPE(dataflow): Finalize the design how capturing/referencing intermediate results is represented.
```
Comment (mavasani) [inline src/Compilers/Core/Portable/Operations/IFlowCaptureReferenceOperation.cs]: We decided to use `CaptureId`. @AlekseyTs let me know if this comment was capturing more work.

File: src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb [hunk 7]
```diff
@@ -1696,7 +1696,6 @@ Namespace Microsoft.CodeAnalysis.Operations
                                                     constantValue,
                                                     isImplicit)
             Else
-                ' PROTOTYPE(dataflow): This code path is not covered by unit-tests
```
Comment (mavasani) [inline src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb]: Covered by added test `SwitchFlow_108`

File: src/Test/Utilities/Portable/Compilation/OperationTreeVerifier.cs [hunk 8]
```diff
@@ -1260,14 +1262,7 @@ public override void VisitObjectCreation(IObjectCreationOperation operation)
         {
             LogString(nameof(IObjectCreationOperation));
 
-            // PROTOTYPE(dataflow): It looks like constructor can be null, when default constructor is used for a structure
-            //                      Apparently we don't have any targeted tests for this case.
```
Comment (mavasani) [inline src/Test/Utilities/Portable/Compilation/OperationTreeVerifier.cs]: Covered by test update above: https://github.com/dotnet/roslyn/pull/27603/files#diff-2522a3cad43b3458befa5cbdea73dfedR1849

File: src/Compilers/VisualBasic/Test/IOperation/IOperation/IOperationTests_ISwitchOperation.vb [hunk 9]
```diff
@@ -9272,5 +9272,79 @@ Block[B10] - Exit
...
+    End Operator
+End Class
+        ]]>
+    </file>
+</compilation>
+
+            Dim compilation = CreateCompilationWithMscorlib40AndVBRuntime(source, parseOptions:=TestOptions.RegularWithFlowAnalysisFeature)
+            compilation.MakeMemberMissing(SpecialMember.System_Nullable_T_GetValueOrDefault)
+
+            compilation.AssertNoDiagnostics()
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Test/IOperation/IOperation/IOperationTests_ISwitchOperation.vb]: >AssertNoDiagnostics [](start = 24, length = 19)

Could you please AssertEmitDiagnostics, I would expect errors for this scenario

**Review by AlekseyTs — APPROVED** [review]
LGTM (iteration 1)

**Review by 333fred — APPROVED** [review]
LGTM (iteration 1)

**Review by AlekseyTs — APPROVED** [review]
LGTM (iteration 2)

**Pull request merged**